addons/sale_project/models/sale_order.py

```diff
@@ -77,24 +79,33 @@ def _compute_tasks_ids(self):
             groupby=['sale_order_id'],
             aggregates=['id:recordset', '__count']
         )
+        completed_tasks_per_so = self.env['project.task']._read_group(
+            domain=['&', ('project_id', '!=', False), '&', ('state', 'in', ['1_done', '1_canceled']), '|', ('sale_line_id', 'in', self.order_line.ids), ('sale_order_id', 'in', self.ids)],
+            groupby=['sale_order_id'],
+            aggregates=['__count']
+        )
```

Comment:
I think this can be done with the previous _read_group, if you group by state too

Reply:
Yes, will do the same